src/devtools/store.js

```diff
@@ -75,6 +75,11 @@ export default class Store extends EventEmitter {
   // When profiling is in progress, operations are stored so that we can later reconstruct past commit trees.
   _isProfiling: boolean = false;
 
+  // These are IDs which were conceptually "removed" from the tree,
+  // but whose Fibers may resurface later either as "add" or "remove" events.
+  // This is what happens to a Suspended "primary" tree in Concurrent Mode.
+  _hiddenIDs: Set<number> = new Set();
```

Comment:
It seems unfortunate that Suspense-specific logic has leaked into the frontend. Is there a way we could track and model this in the renderer so that the `Store` and `CommitTreeBuilder` don't need to be aware of this particular quirk of Suspense?

Reply:
To clarify: Maybe we could track the same set of ids in the backend, and send (or not send) mount/unmount events for the hidden fibers (not sure which would be most appropriate since I haven't really digested the essence of this fix yet).